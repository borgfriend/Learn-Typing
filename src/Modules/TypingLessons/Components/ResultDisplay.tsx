import * as React from 'react';

import { observer } from 'mobx-react';
import { exerciseStore } from '../Stores/ExerciseStore';
import { Button } from '@blueprintjs/core';

export const ResultDisplay: React.SFC = observer(() => (
    <>
        <ul>
            <li>Mistakes: {exerciseStore.mistakes}</li>
            <li>Time: {exerciseStore.time}</li>
        </ul>
        <Button onClick={() => { exerciseStore.reset(); }}>Reset</Button>
    </>
));