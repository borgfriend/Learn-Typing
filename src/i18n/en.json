{
  "APP_TITLE": "Learn Typing",
  "START_LESSON": "Start Lesson"
}