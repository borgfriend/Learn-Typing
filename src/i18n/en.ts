export const i18n = {
    Reset: "Reset",
    Mistakes: "Mistakes",
    Time: "Time"
}