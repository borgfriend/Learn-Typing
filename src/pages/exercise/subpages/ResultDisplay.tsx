import * as React from "react";

import { Button } from "../../../components/Button/Button";
import { i18n } from "../../../i18n/en";
import { LessonStats } from "../TypingLesson";

export const ResultDisplay: React.FC<{
  lessonStats: LessonStats;
  resetAction: () => void;
}> = ({ lessonStats, resetAction }) => {
  const { mistakes, time } = lessonStats;
  return (
    <>
      <ul>
        <li>{i18n.Mistakes}: {mistakes}</li>
        <li>{i18n.Time}: {time}</li>
      </ul>
      <Button onClick={resetAction}>{i18n.Reset}</Button>
    </>
  );
};
