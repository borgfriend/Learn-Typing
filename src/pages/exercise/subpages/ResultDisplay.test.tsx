import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { i18n } from "../../../i18n/en";
import { ResultDisplay } from "./ResultDisplay";

it("renders a default button", () => {
  const mockFn = jest.fn();
  const { container, getByText } = render(
    <ResultDisplay
      lessonStats={{ mistakes: 0, time: 0 }}
      resetAction={mockFn}
    ></ResultDisplay>
  );
  expect(container).toMatchSnapshot();
  const resetButton = getByText(i18n.Reset);
  fireEvent.click(resetButton);
  expect(mockFn).toHaveBeenCalled();
});
