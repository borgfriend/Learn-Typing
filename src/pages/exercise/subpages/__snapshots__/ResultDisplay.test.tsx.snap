// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a default button 1`] = `
<div>
  <ul>
    <li>
      Mistakes
      : 
      0
    </li>
    <li>
      Time
      : 
      0
    </li>
  </ul>
  <button
    class="sc-bczRLJ fiMtFI"
  >
    Reset
  </button>
</div>
`;
